pallets/relayer/src/lib.rs

```diff
@@ -193,11 +195,9 @@ impl<T: Config> Module<T> {
     }
 
     fn base_accept_paying_key(origin: T::Origin, auth_id: u64) -> DispatchResult {
-        let PermissionedCallOriginData {
-            sender: user_key,
-            primary_did: user_did,
-            ..
-        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
```

Comment:
Not checking perms does seem strange to me, so we should review why this is to be called with the paying DID.
Going by the current code, it seems like we won't even check CDD for `user_key`'s DID, since that is not `user_did`.

Reply:
The current logic does check that the `user_key` has a CDD.  If we want to change back to using `ensure_origin_call_permissions`, then we will need to change the logic in the `CddHandler` in `runtime/*/src/fee_details.rs` to not change the current identity when `accept_paying_key` is called.